packages/fastify/modules/plugin/plugin.js

```diff
@@ -0,0 +1,82 @@
+const fastifyPlugin = require('fastify-plugin');
+const merge = require('merge-options');
+const {
+  Prometheus,
+  createMetricTypes,
+  createRequestRecorder,
+  createGcObserver,
+  defaultNormalizers,
+  getSummary,
+  getContentType,
+} = require('@promster/metrics');
+const pkg = require('../../package.json');
+
+const defaultRouteOptions = {
+  route: { method: 'GET', url: '/metrics' },
+};
+
+let recordRequest;
+let upMetric;
+
+const extractPath = req => req.raw.originalUrl || req.raw.url;
+const getRequestRecorder = () => recordRequest;
+const signalIsUp = () => upMetric && upMetric.set(1);
+const signalIsNotUp = () => upMetric && upMetric.set(0);
+
+const createPlugin = async (fastify, options) => {
+  const defaultedOptions = merge(
+    createMetricTypes.defaultedOptions,
+    createRequestRecorder.defaultedOptions,
+    defaultNormalizers,
+    defaultRouteOptions,
+    options
+  );
+
+  const metricTypes = createMetricTypes(defaultedOptions);
+  const observeGc = createGcObserver(metricTypes);
+
+  recordRequest = createRequestRecorder(metricTypes, defaultedOptions);
+  upMetric = metricTypes && metricTypes.up;
+
+  observeGc();
+
+  fastify.decorate('Prometheus', Prometheus);
+  fastify.decorate('recordRequest', recordRequest);
+  fastify.decorateRequest('__promsterStartTime__', null);
+
+  fastify.addHook('onRequest', async (req, _) => {
+    req.__promsterStartTime__ = process.hrtime();
+  });
+
+  fastify.addHook('onResponse', async (req, reply) => {
+    recordRequest(req.__promsterStartTime__, {
+      labels: {
+        method: defaultedOptions.normalizeMethod(req.raw.method),
+        // eslint-disable-next-line camelcase
+        status_code: defaultedOptions.normalizeStatusCode(reply.res.statusCode),
+        path: defaultedOptions.normalizePath(extractPath(req)),
+        ...(defaultedOptions.getLabelValues &&
+          defaultedOptions.getLabelValues(req, reply)),
+      },
+    });
+  });
+
+  fastify.route({
```

Comment:
This I will remove. I am aware that this is a bit controversal but I would like to 

1. Enforce best practices: the Prometheus server should be another server aside the app
   - This makes it run on another port (when using `@promster/server`) and makes it easier to not expose in a k8s setup
2. Give people more flexibility has to how to expose it with very few lines of code if they want to